Domains: Include WPCOM email in the messaging for WHOIS update on WWD domains

klimeryk opened the pull request:
There's been a change in how the user can confirm a WHOIS update for a WWD domain (backend changes merged, see `D4312-code`), we now send a third, backup email to their WPCOM account.

So, up to three emails will be sent:

1. If the email has not been changed, the confirmation email will be sent to the current email plus the WPCOM one.
   <img width="1106" alt="screen shot 2017-02-28 at 19 18 08" src="https://cloud.githubusercontent.com/assets/3392497/23480488/67260cd4-fec8-11e6-98ae-018c8982043c.png">


2. If the email has been changed, the confirmation email will be sent to the current email (with the WPCOM one acting as a backup) and the new email.
   <img width="1158" alt="screen shot 2017-02-28 at 19 18 53" src="https://cloud.githubusercontent.com/assets/3392497/23480501/6fc2c3d2-fec8-11e6-834d-4f37168a8e30.png">


I've updated the copy a bit to mention that we'll be sending this backup email to their WPCOM email address. If the current email and WPCOM one are the same, we don't mention them both, since only one email will be sent in that case.

_Added:_
I've also updated the logic to not show the above dialog if we did not change any of the main fields (first name, last name, organization, email) - because in that case, no confirmation is needed, even in WWD case ;)

umurkontaci commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -46,7 +47,8 @@ class EditContactInfoFormCard extends React.Component {
 			React.PropTypes.bool
-		] ).isRequired
+		] ).isRequired,
+		currentUser: React.PropTypes.object.isRequired
```
Can `currentUser` be null if there's a race?

umurkontaci commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -118,6 +120,26 @@ class EditContactInfoFormCard extends React.Component {
 
+	hasEmailChanged() {
+		return this.props.contactInformation.email === formState.getFieldValue( this.state.form, 'email' );
+	}
```
I'd prefer something like `isEmailSame`, `hasXXXChanged` sounds like this function is tracking a pristine/dirty state for the form whereas it's just an equality check.

klimeryk replied:
`hasEmailChanged` just sounds more proper to me - I prefer to write code so that it reads as much as a proper sentence as possible. `isEmailSame` feels awkward to me.
And, actually, this function does check if the form's state has changed from its original value.

umurkontaci commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -118,6 +120,26 @@ class EditContactInfoFormCard extends React.Component {
+		return this.props.translate( '%(currentEmail)s (additionally to %(wpcomEmail)s)',
+			{
+				args: { currentEmail, wpcomEmail },
+				context: 'List of emails the WHOIS confirmation email is sent to'
```
I'm not a big fan of the copy here, it looks like we are making an already long text longer 

/cc: @breezyskies @ranh ( @Automattic/cerulean ) would love a quick copy input if you have the time

breezyskies replied:
I don't see this copy in the screenshots above. Do you have a screenshot of where it shows up?

klimeryk replied:
@breezyskies, it's in both screenshots above :) It's just part of the larger sentence. It's constructed separately since we mention one email (if they are the same) and two if the wpcom one differs (so the wpcom one becomes the backup email). See `p7DVsv-1TW-p2` for more details :)

a8ci18n commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -118,6 +120,26 @@ class EditContactInfoFormCard extends React.Component {
+			return currentEmail;
+		}
+
+		return this.props.translate( '%(currentEmail)s (additionally to %(wpcomEmail)s)',
```
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

a8ci18n replied:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

a8ci18n commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -35,6 +35,7 @@ import support from 'lib/url/support';
 import { registrar as registrarNames } from 'lib/domains/constants';
 import DesignatedAgentNotice from 'my-sites/upgrades/domain-management/components/designated-agent-notice';
 import Dialog from 'components/dialog';
+import { getCurrentUser } from 'state/current-user/selectors';
```
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Emails have been sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. Please ensure they're both confirmed to finish this process.' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n replied:
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'To finish this process, this change will need to be confirmed by emails sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. Please make sure you'll be able to do so.' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n replied:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

a8ci18n commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -420,25 +443,26 @@ class EditContactInfoFormCard extends React.Component {
 							}
 						);
 					} else {
+						const newEmail = formState.getFieldValue( this.state.form, 'email' );
```
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Emails have been sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. Please ensure they're both confirmed to finish this process.' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/third-email-whois-update)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n replied:
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Emails have been sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. Please ensure they're both confirmed to finish this process.' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/third-email-whois-update)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -167,21 +189,22 @@ class EditContactInfoFormCard extends React.Component {
 				}
 			);
 		} else {
+			const newEmail = formState.getFieldValue( this.state.form, 'email' );
```
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'To finish this process, this change will need to be confirmed by emails sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. Please make sure you'll be able to do so.' )` *ES Score: 13*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/third-email-whois-update)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n replied:
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'To finish this process, this change will need to be confirmed by emails sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. Please make sure you'll be able to do so.' )` *ES Score: 13*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/third-email-whois-update)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -118,6 +120,26 @@ class EditContactInfoFormCard extends React.Component {
+		if ( currentEmail === wpcomEmail ) {
+			return currentEmail;
+		}
+
```
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

a8ci18n replied:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

a8ci18n commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -420,25 +443,26 @@ class EditContactInfoFormCard extends React.Component {
+
 						message = this.props.translate(
-							'Emails have been sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. ' +
+							'Emails have been sent to {{strong}}%(currentEmails)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. ' +
```
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Emails have been sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. Please ensure they're both confirmed to finish this process.' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/third-email-whois-update)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -167,21 +189,22 @@ class EditContactInfoFormCard extends React.Component {
 		} else {
+			const newEmail = formState.getFieldValue( this.state.form, 'email' );
+
 			text = translate( 'To finish this process, this change will need to be confirmed by emails ' +
```
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'To finish this process, this change will need to be confirmed by emails sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. Please make sure you'll be able to do so.' )` *ES Score: 13*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/third-email-whois-update)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

klimeryk replied:
This string was not changed, so no need to alert me about it, @a8ci18n 😞

umurkontaci replied:
Apart from the bug, I think it's a valuable improvement. Downvoting will have a negative affect on the score of the suggestion

yoavf replied:
@klimeryk is it possible that this comment was made when the string was being changed? In any case this comment will not be posted again if there's another commit or a rebase of this pr

klimeryk replied:
@yoavf That was partly my fault, party GitHub's way of showing this change. First of all - IIRC, the string _did_ change, just on a line below - that was the continuation of this string. So, @a8ci18n was right, it's just from the small context that GitHub showed it, it seemed like a bug. So, sorry for the false alarm 🙇

ranh commented:
@klimeryk @umurkontaci @breezyskies How about:

First case:

> We’ll email you at old@example.com and new@example.com with a link to confirm the new details. The change won’t go live until we receive confirmation from both emails.
> 
> If you don’t have access to old@example.com, we will also email you at wpcom@example.com, as backup.

Second case:

> We’ll email you at old@example.com and wpcom@example.com with a link to confirm the new details. The change won't go live until we receive confirmation from one of these emails. 

I would also suggest we change the "Confirm" button label to something else, perhaps "Request Confirmation".

klimeryk commented:
Copy updated - I've updated the screenshots in the first post and added a small change I've added. I'm checking if any of the main WHOIS fields changed (the ones that trigger these confirmation emails) - if they did not change we should not show these dialogs, just save the record (no confirmation is needed then).

umurkontaci commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -152,6 +170,20 @@ class EditContactInfoFormCard extends React.Component {
+			'If you don’t have access to {{strong}}%(currentEmail)s{{/strong}}, ' +
+			'we will also email you at {{strong}}%(wpcomEmail)s{{/strong}}, as backup.', {
+				args: { currentEmail, wpcomEmail },
+				components: { supportLink: <a href={ support.CALYPSO_CONTACT } />, strong }
```
We could extract `supportLink` to a const or inline `strong` so they are uniform.

klimeryk replied:
Actually, `supportLink` is not used anymore in that string, so it can be removed 🎉

umurkontaci commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -162,45 +194,44 @@ class EditContactInfoFormCard extends React.Component {
+			text = translate(
+				'We’ll email you at {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}} ' +
+				'with a link to confirm the new details. The change won’t go live until we receive confirmation from both emails.',
+				{ args: { oldEmail: currentEmail, newEmail }, components: { strong } }
```
s/oldEmail/currentEmail/

klimeryk replied:
Same as https://github.com/Automattic/wp-calypso/pull/11485#discussion_r103903922

umurkontaci commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -406,44 +437,38 @@ class EditContactInfoFormCard extends React.Component {
+					'Emails have been sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. ' +
+					'Please ensure they\'re both confirmed to finish this process.',
+					{
+						args: { oldEmail: currentEmail, newEmail },
```
s/oldEmail/currentEmail/

klimeryk replied:
That was my initial idea, but @a8ci18n bot suggested to revert it because we have this string translated in 17 languages. Since the copy did not actually change, it makes sense to keep the placeholder the same so that we don't loose those translations.

umurkontaci replied:
ah, good to know!

umurkontaci commented:
I've had couple small code style suggestions, but otherwise LGTM

a8ci18n commented on client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx:
```diff
@@ -162,45 +194,44 @@ class EditContactInfoFormCard extends React.Component {
 				{
 					action: 'confirm',
-					label: this.props.translate( 'Confirm' ),
+					label: this.props.translate( 'Request Confirmation' ),
```
Hi! I've found a possible matching string that has already been translated **45** times:
`translate( 'Confirmation' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).
